Assets/Scripts/Buildings/ConstructionSite/ResourceTextHandler.cs

```diff
@@ -0,0 +1,31 @@
+using TMPro;
+using UnityEngine;
+
+namespace Cosmobot
+{
+    public class ResourceTextHandler : MonoBehaviour
+    {
+        [SerializeField] private Transform text;
+        [SerializeField] private TextMeshPro displayText;
+        private new Camera camera;
+        void Start()
+        {
+            camera = Camera.main;
+        }
+
+        void Update()
+        {
+            text.LookAt(camera.transform);
+            text.rotation = Quaternion.LookRotation(camera.transform.forward);
+        }
+
+        public void InitializeText(int resourceAmount) {
+            displayText.text = resourceAmount.ToString();
+            UpdateText(resourceAmount);
+        }
+
+        public void UpdateText(int newAmount) {
+            displayText.text = $"{newAmount}";
+        }
```

Comment:
idk, but i don't think this will have any impact on performance. `UpdateText` is called only when Resource amount changes. So in worst case scenario - one time per frame and at most ~10 times per building